cmd/apply.go

```diff
@@ -484,97 +465,171 @@ func createFunctionsAuth() error {
 	}
 
-	log.Println(res.ExitCode, res.Stdout, res.Stderr)
+	if res.ExitCode != 0 {
+		return fmt.Errorf("non-zero exit-code: %s %s", res.Stdout, res.Stderr)
+	}
 
+	if len(res.Stderr) > 0 {
+		log.Printf("stderr: %s\n", res.Stderr)
+	}
 	return nil
 }
 
 func installSealedSecrets() error {
-	log.Println("Creating SealedSecrets")
+	log.Println("Installing sealed-secrets")
+
+	var env []string
+	args := []string{"install", "sealed-secrets", "--namespace=kube-system", "--wait"}
 
 	task := execute.ExecTask{
-		Command:     "scripts/install-sealedsecrets.sh",
+		Command:     "arkade",
+		Args:        args,
 		Shell:       true,
-		StreamStdio: true,
+		Env:         env,
+		StreamStdio: false,
 	}
 
-	taskRes, err := task.Execute()
-
+	res, err := task.Execute()
 	if err != nil {
 		return err
 	}
 
-	if len(taskRes.Stderr) > 0 {
-		log.Println(taskRes.Stderr)
+	if res.ExitCode != 0 {
+		return fmt.Errorf("non-zero exit-code: %s %s", res.Stdout, res.Stderr)
 	}
 
+	if len(res.Stderr) > 0 {
+		log.Printf("stderr: %s\n", res.Stderr)
+	}
 	return nil
 }
 
-func installOpenfaas(scaleToZero, ingressOperator bool, additionalPaths []string) error {
-	log.Println("Creating OpenFaaS")
+func installOpenfaas(scaleToZero, ingressOperator bool) error {
+	log.Println("Installing openfaas")
+
+	args := []string{"install", "openfaas",
+		"--set basic_auth=true",
+		"--set functionNamespace=openfaas-fn",
+		"--set ingress.enabled=false",
+		"--set gateway.scaleFromZero=true",
+		"--set gateway.readTimeout=15m",
+		"--set gateway.writeTimeout=15m",
+		"--set gateway.upstreamTimeout=14m55s",
+		"--set queueWorker.ackWait=15m",
+		"--set faasnetes.readTimeout=5m",
+		"--set faasnetes.writeTimeout=5m",
+		"--set gateway.replicas=2",
+		"--set queueWorker.replicas=2",
+		"--set faasIdler.dryRun=" + strconv.FormatBool(!scaleToZero),
+		"--set faasnetes.httpProbe=true",
+		"--set faasnetes.imagePullPolicy=IfNotPresent",
+		"--set ingressOperator.create=" + strconv.FormatBool(ingressOperator),
+		"--wait",
+	}
 
 	task := execute.ExecTask{
-		Command: "scripts/install-openfaas.sh",
-		Shell:   true,
-		Env: []string{
-			fmt.Sprintf("FAAS_IDLER_DRY_RUN=%v", strconv.FormatBool(!scaleToZero)),
-			fmt.Sprintf("INSTALL_INGRESS_OPERATOR=%v", strconv.FormatBool(ingressOperator)),
-		},
-		StreamStdio: true,
+		Command:     "arkade",
+		Args:        args,
+		Shell:       true,
+		StreamStdio: false,
 	}
 
 	res, err := task.Execute()
-
 	if err != nil {
 		return err
 	}
 
-	log.Println(res.ExitCode, res.Stdout, res.Stderr)
+	if res.ExitCode != 0 {
+		return fmt.Errorf("non-zero exit-code: %s %s", res.Stdout, res.Stderr)
+	}
+
+	if len(res.Stderr) > 0 {
+		log.Printf("stderr: %s\n", res.Stderr)
+	}
 
 	return nil
 }
 
-func installMinio() error {
-	log.Println("Creating Minio")
+func getS3Credentials() (string, string, error) {
+	var accessKey, secretKey string
+
+	args := []string{"get", "secret", "-n", "openfaas", "s3-access-key", "-o jsonpath='{.data.s3-access-key}'"}
```

Comment:
These secrets live in the openfaas-fn namespace

Reply:
I'll change it, thanks for catching the problem.